org.eclipse.jdt.ls.product/scripts/jdtls.py

```diff
@@ -0,0 +1,102 @@
+###############################################################################
+# Copyright (c) 2022 Marc Schreiber and others.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+# Marc Schreiber - initial API and implementation
+###############################################################################
+import argparse
+import os
+import platform
+import re
+import subprocess
+from pathlib import Path
+
+def get_java_executable(validate_java_version):
+	java_executable = 'java'
+
+	if 'JAVA_HOME' in os.environ:
+		java_exec_to_test = Path(os.environ['JAVA_HOME']) / 'bin' / 'java'
+		if java_exec_to_test.is_file():
+			java_executable = java_exec_to_test.resolve()
+
+	if not validate_java_version:
+		return java_executable
+
+	out = subprocess.check_output([java_executable, '-version'], stderr = subprocess.STDOUT, universal_newlines=True)
+
+	matches = re.finditer(r"(?P<major>\d+)\.\d+\.\d+", out)
+	for match in matches:
+		java_major_version = int(match.group("major"))
+
+		if java_major_version < 11:
+			raise Exception("jdtls requires at least Java 11")
+
+		return java_executable
+
+	raise Exception("Could not determine Java version")
+
+def find_equinox_launcher(jdtls_base_directory):
+	plugins_dir = jdtls_base_directory / "plugins"
+	launchers = plugins_dir.glob('org.eclipse.equinox.launcher_*.jar')
+	for launcher in launchers:
+		return plugins_dir / launcher
+
+	raise Exception("Cannot find equinox launcher")
+
+def get_shared_config_path(jdtls_base_path):
+	system = platform.system()
+
+	if system == 'Linux':
+		config_dir = 'config_linux'
+	elif system == 'Darwin':
+		config_dir = 'config_mac'
+	elif system == 'Windows':
+		config_dir = 'config_win'
+	else:
+		raise Exception("Unknown platform {} detected".format(platform))
+
+	return jdtls_base_path / config_dir
+
+def main(args):
+	parser = argparse.ArgumentParser()
+	parser.add_argument("--validate-java-version", default=True, action=argparse.BooleanOptionalAction)
```

Comment:
`argparse.BooleanOptionalAction` is only available in python 3.9. So essentially everyone using python version < 3.9, get the above error.

Reply:
Issue has been raised in #2153